### ISSUE TITLE
reducer cannot be used in composed contracts

### ISSUE THREAD

**Issue by Trivo25** [o1-labs/o1js#359]
`reducer.reduce` cannot be used within a contract that has been called by a caller contract

`Error: Can not analyze methods inside Circuit.runAndCheck, because this creates a circuit nested in another circuit`

MRE

```ts
class ReducerContract extends SmartContract {
  reducer = Experimental.Reducer({ actionType: Field });

  @method reduceEvents(): Field {

    let { state: s, actionsHash: a } = this.reducer.reduce(
      [[Field.zero]],
      Field,
      (state: Field, _action: Field) => {
        return _action;
      },
      // initial state
      { state: Field.zero, actionsHash: Field.zero }
    );
    return Field.zero;
  }
}

class Caller extends SmartContract {
  @state(Field) sum = State<Field>();
  events = { sum: Field };

  @method callAddAndEmit() {
    let c = new ReducerContract(adderAddress);
    let sum = c.reduceEvents();
  }
}

// ...

tx = await Mina.transaction(feePayer, () => {
  zkapp.callAddAndEmit();
});
```

https://github.com/o1-labs/snarkyjs/blob/main/src/lib/zkapp.ts#L685

# Temporary fix to unblock integration tests
- [x] pass in dummy values instead of invoking the function that causes the issue https://github.com/o1-labs/snarkyjs/blob/feature%2Fvoting-contract/src/lib/zkapp.ts#L769

# TODO
- [x] remove dummy values
- [x] temporary fix: call `.compile`

# Update

`analyzeMethods` behaves weirdly sometimes, it would be nice to analyze all contracts that are part of a tx in advance, but is out of scope for now

**Comment by Trivo25** [comment]
This only appears when compile *was not* invoked. Solution: Make sure `analyzeMethod` is being invoked anyway

**Comment by mitschabaude** [comment]
Yeah that's a partial solution at least!
